core/src/test/java/org/infinispan/functional/FunctionalSimpleTutorialTest.java

```diff
@@ -0,0 +1,83 @@
+package org.infinispan.functional;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.api.functional.EntryView.ReadEntryView;
+import org.infinispan.commons.api.functional.FunctionalMap.ReadOnlyMap;
+import org.infinispan.commons.api.functional.FunctionalMap.ReadWriteMap;
+import org.infinispan.commons.api.functional.FunctionalMap.WriteOnlyMap;
+import org.infinispan.commons.api.functional.MetaParam.MetaLifespan;
+import org.infinispan.commons.api.functional.Traversable;
+import org.infinispan.functional.impl.FunctionalMapImpl;
+import org.infinispan.functional.impl.ReadOnlyMapImpl;
+import org.infinispan.functional.impl.ReadWriteMapImpl;
+import org.infinispan.functional.impl.WriteOnlyMapImpl;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+@Test(groups = "functional", testName = "functional.FunctionalSimpleTutorialTest")
+public class FunctionalSimpleTutorialTest extends SingleCacheManagerTest {
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      return TestCacheManagerFactory.createCacheManager();
+   }
+
+   public void testSimpleTutorial() throws Exception {
+      AdvancedCache<String, String> cache = cacheManager.<String, String>getCache().getAdvancedCache();
+      FunctionalMapImpl<String, String> functionalMap = FunctionalMapImpl.create(cache);
+      WriteOnlyMap<String, String> writeOnlyMap = WriteOnlyMapImpl.create(functionalMap);
+      ReadOnlyMap<String, String> readOnlyMap = ReadOnlyMapImpl.create(functionalMap);
+
+      // Execute two parallel write-only operation to store key/value pairs
+      CompletableFuture<Void> writeFuture1 = writeOnlyMap.eval("key1", "value1",
+         (v, writeView) -> writeView.set(v));
+      CompletableFuture<Void> writeFuture2 = writeOnlyMap.eval("key2", "value2",
+         (v, writeView) -> writeView.set(v));
+
+      // When each write-only operation completes, execute a read-only operation to retrieve the value
+      CompletableFuture<String> readFuture1 =
+         writeFuture1.thenCompose(r -> readOnlyMap.eval("key1", ReadEntryView::get));
+      CompletableFuture<String> readFuture2 =
+         writeFuture2.thenCompose(r -> readOnlyMap.eval("key2", ReadEntryView::get));
+
+      // When the read-only operation completes, print it out
+      System.out.printf("Created entries: %n");
```

Comment:
I have a script that collapses consecutive "Test suite progress:" lines in the console, so I guess it's more visible for me... but I don't think having the logging different in the test would cause many issues either :)